tidy.Rmd

```diff
@@ -124,54 +124,120 @@ Typically a dataset will only suffer from one of these problems; it'll only suff
 
 * The set of columns whose names are values, not variables. In this example, 
   those are the columns `1999` and `2000`.
 
-* The name of the variable to move the column names to. Here it is `year`.
+* The name of the variable to move the column names to: `year`.
 
-* The name of the variable to move the column values to. Here it's `cases`.
+* The name of the variable to move the column values to: `cases`.
   
 Together those parameters generate the call to `pivot_longer()`:
 
 ```{r}
 table4a %>% 
-  pivot_longer(c(`1999`, `2000`), names_to = "year", values_to = "cases")
+  pivot_longer(
+    cols = c(`1999`, `2000`),
+    names_to = "year", 
+    values_to = "cases"
+    )
 ```
 
-The columns to pivot are specified with `dplyr::select()` style notation. Here there are only two columns, so we list them individually. Note that "1999" and "2000" are non-syntactic names (because they don't start with a letter) so we have to surround them in backticks. To refresh your memory of the other ways to select columns, see [select](#select).
+The columns to pivot are specified with `dplyr::select()` style notation in the `cols` argument. Here there are only two columns, so we list them individually. Note that `1999` and `2000` are non-syntactic names (because they don't start with a letter) so we have to surround them in backticks. To refresh your memory of the other ways to select columns, see [select](#select).
+
+`year` and `cases` do not exist in `table4a` so we put their names in quotes in `names_to` and `values_to` arguments, respectively.
 
-`year` and `cases` do not exist in `table4a` so we put their names in quotes.
+In the final result, the pivoted columns are dropped, and we get new `year` and `cases` columns. Otherwise, the relationships between the original variables are preserved. Visually, this is shown in Figure \@ref(fig:tidy-pivot-longer). 
 
-```{r tidy-gather, echo = FALSE, out.width = "100%", fig.cap = "Pivoting `table4` into a longer, tidy form."}
+```{r tidy-pivot-longer, echo = FALSE, out.width = "100%", fig.cap = "Pivoting `table4a` into a \"longer\", tidy form."}
 knitr::include_graphics("images/tidy-9.png")
 ```
 
-In the final result, the pivoted columns are dropped, and we get new `year` and `cases` columns. Otherwise, the relationships between the original variables are preserved. Visually, this is shown in Figure \@ref(fig:tidy-gather). 
+There is still one issue though. Take a peek at the type of the `year` variable. We would expect `year` to be numeric (or specifically, we would expect it to be an integer), however it's showing up as a character. This is because the values in the `year` variable came from column headings in `table4a`. We can transform the type of this variable as we're pivoting the data using the `names_transform` argument in `pivot_longer()`. In this case we don't also need to transform the type of the `cases` variable, but if we did, we could simply add that to the list in `names_transform` as well.
```

Comment:
Do you think it would be better to just use `mutate()` here? And call back to https://r4ds.had.co.nz/data-import.html#parsing-a-vector for other functions?

Reply:
I think showing the `mutate()` option first is better, and then we could show you can achieve the same with `names_transform()`. Keeps it simple for those who want to stick to what they know but also points to additional `pivot_*()` functionality for the more adventurous reader. Does that sound good?

Reply:
I'm not sure I like `names_transform` enough to even mention it here.

Reply:
I am all for keeping it to just using `mutate()`. I think that feels like a more natural way to build a pipeline. 

I would suggest the same change further down in the case study section too then, just taking out `names_transform()` altogether and accomplishing its purpose with `pivot_*() %>% mutate()`.

Reply:
Yeah, agreed.

Reply:
Alright, all these changes are implemented now.